staging/src/k8s.io/csi-translation-lib/plugins/azure_disk.go

```diff
@@ -99,6 +100,9 @@ func (t *azureDiskCSITranslator) TranslateInTreeInlineVolumeToCSI(volume *v1.Vol
 	}
 
 	azureSource := volume.AzureDisk
+	if azureSource.Kind != nil && !strings.EqualFold(string(*azureSource.Kind), managed) {
```

Comment:
it's not clear to me why this comparison is being done case-insensitively... API validation of AzureDiskVolumeSource requires the value to be a valid enum:

```
	if azure.Kind != nil && !supportedDiskKinds.Has(string(*azure.Kind)) {
		allErrs = append(allErrs, field.NotSupported(fldPath.Child("kind"), *azure.Kind, supportedDiskKinds.List()))
	}
```